### PR TITLE
BaSD-MS-19-Clients: Create react logic

### PR DESCRIPTION
**Pull request by ignacioaristo**
### Resume:
Frontend was connected with backend using React.

**Methods developed for Clients:**

- Get all the Clients

- Create a Client

> Click on Add Client button to display the form.

- Update a Client

> Click on the row of the Client to display the form and rewrite it.

- Delete a Client

> Click on the trash can icon to delete the Client.

Modal was created for delete option.

Validations were added to every form.

### CODE REVIEW

File: src/Components/Clients/Form/FormClients.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    fetch(url, options).then((response) => {
+      if (response.status !== 200 && response.status !== 201){
+        return response.json().then(({message}) => {
+          throw new Error(message);
+        });
+      }
+      return response.json();
+    })
+    .catch((error) => {
+      return error;
```
Comment (aquilesmariani) [inline src/Components/Clients/Form/FormClients.js:99]: It doesn't have much sense to just return the error, let's show it into the UI

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
We should consider keeping `package-lock.json` file as well, I don't know if you meant to delete it but we need to keep track of it in git

File: .eslintrc.json [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+      "version": "detect"
     }
-}
+  },
+  "env": {
+    "browser": true,
+    "node": true,
+    "es2021": true
+  },
+  "extends": ["eslint:recommended", "plugin:react/recommended" /*, "plugin:prettier/recommended"*/],
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply .eslintrc.json:12]: Yes i did, now were removed.

File: .eslintrc.json [hunk 3]
```diff
@@ -1,27 +1,33 @@
...
+    },
+    "ecmaVersion": 12,
+    "sourceType": "module"
+  },
+  "plugins": ["react"],
+  "rules": {
+    "no-unused-vars": "warn",
+    "semi": [2, "always"],
+    "react/react-in-jsx-scope": "off",
+    // "prettier/prettier": [
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply .eslintrc.json:25]: Fixed.

File: src/Components/Clients/Form/FormClients.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  const onChangeOpenPositions = (event) => {
+    setOpenPositionsValue(event.target.value);
+  };
+
+  const params = new URLSearchParams(window.location.search);
+  const clientId = params.get('id');
+
+  useEffect(()=> {
+    if(clientId){
+      fetch(`${process.env.REACT_APP_API}/api/clients/id/${clientId}`)
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply src/Components/Clients/Form/FormClients.js:40]: Fixed it.

File: src/Components/Clients/Form/FormClients.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        .then((response) => {
+          if (response.status !== 200) {
+            return response.json().then(({ message }) => {
+              throw new Error(message);
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply src/Components/Clients/Form/FormClients.js:48]: Modal was added.

File: src/Components/Clients/index.js [hunk 6]
```diff
@@ -1,24 +1,89 @@
...
+    fetch(url, {
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    })
+      .then((res) => {
+        if (res.status !== 204) {
+          return res.json().then((message) => {
+            throw new Error(message);
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply src/Components/Clients/index.js:41]: Modal was added.

File: src/Components/Clients/Form/FormClients.js (hunk 1 above)
Comment (ignacioaristo) [reply src/Components/Clients/Form/FormClients.js]: Modal was added.

File: src/Components/Clients/Form/FormClients.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        email: emailValue,
+        phone: phoneValue,
+        openPositions: openPositionsValue
+      })
+    };
+
+    if (clientId === null) {
+      options.method = 'POST';
+      url = `${process.env.REACT_APP_API}/clients/add`;
+      window.location.href = `/clients`;
```
Comment (aquilesmariani) [inline src/Components/Clients/Form/FormClients.js:91]: This is redirecting you at this moment, let's put this after the request when it's successful

File: src/Components/Clients/Form/FormClients.js [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+    } else {
+      options.method = 'PUT';
+      url = `${process.env.REACT_APP_API}/clients/update/${clientId}`;
+      window.location.href = `/clients`;
```
Comment (aquilesmariani) [inline src/Components/Clients/Form/FormClients.js:95]: Same as above

File: src/Components/Clients/index.js [hunk 9]
```diff
@@ -1,24 +1,123 @@
...
+          <th>Name</th>
+          <th>Company Type</th>
+          <th>Email</th>
+          <th>Country</th>
+          <th>Phone</th>
+          <th>Actions</th>
+        </thead>
+        <tbody>
+          {clients.map((client) => {
+            return (
```
Comment (aquilesmariani) [inline src/Components/Clients/index.js:94]:
```suggestion
          {clients.map((client) => (
```

File: src/Components/Clients/Modal/modalClient.module.css [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+.modal button {
+  width: 100px;
+  height: 50px;
+  padding: 5px 15px;
+  margin: 20px;
+  border: none;
+  border-radius: 7px;
+  background-color: #002147;
+  color: white;
```
Comment (aquilesmariani) [inline src/Components/Clients/Modal/modalClient.module.css:31]: Let's start putting colors with hex so we keep consistency across the CSS files

**Review by aquilesmariani — CHANGES_REQUESTED** [review]
Almost there!

File: src/Components/Clients/Modal/modalClient.module.css (hunk 10 above)
Comment (ignacioaristo) [reply src/Components/Clients/Modal/modalClient.module.css:31]: Changed

File: src/Components/Clients/Form/FormClients.js (hunk 7 above)
Comment (ignacioaristo) [reply src/Components/Clients/Form/FormClients.js:91]: Modal was created for error message

File: src/Components/Clients/Form/FormClients.js (hunk 8 above)
Comment (ignacioaristo) [reply src/Components/Clients/Form/FormClients.js:95]: Fixex

File: src/Components/Clients/index.js (hunk 9 above)
Comment (ignacioaristo) [reply src/Components/Clients/index.js:94]: Return removed

**Review by aquilesmariani — APPROVED** [review]
Looks great!

**Pull request merged**